extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java

```diff
@@ -612,6 +613,9 @@ public void serverSerializers(ResteasyReactiveRecorder recorder,
         }
 
         serverSerializersProducer.produce(new ServerSerialisersBuildItem(serialisers));
+
+        // conditional serialisers
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, MultipartMessageBodyWriter.class.getName()));
```

Comment:
Why are we always registering this for reflection even if it's not going to be used?

Reply:
I tried to register conditionally in here: QuarkusServerEndpointIndexer.handleMultipartForReturnType, but didn't work. Let me try again.

Reply:
It should be fixed now.